### DISCUSSION TITLE
change the experiment background color from white to gray

### DISCUSSION THREAD

**Discussion by ontinglo** [jspsych/jsPsych#936] [Q&A] [answered]
hi guys 

i am trying to modify the codes in the jspsych.css to change the background from white to gray when running the experiments, i change the Hex Code to #808080 in different areas but i still couldn't make it

may i know the exact location in the jspsych.css for changing the background color?

thanks a lot!

onting

**Comment by psychbruce** [comment, marked as answer]
You can add this in a CSS file:
```css
body { background-color: #808080; }
```

But a better way is to set HTML DOM style by using the "document" object in a JavaScript file, like this:
```javascript
document.body.style.backgroundColor = '#808080'
```

For more useful examples (demos) of jsPsych experiments, see:
https://github.com/psychbruce/jspsych

**Reply by ontinglo** [reply]
thank you so much bruce it works

**Comment by becky-gilbert** [comment]
One other thing to note is that, when there are conflicting CSS rules, the browser has a system for determining which rule should apply. [This page](https://www.techrepublic.com/blog/web-designer/css-specificity-hierarchy-what-you-need-to-know/) might help. 
Also, if there are different CSS rules specified for the same CSS selector (e.g. body { background-color: #808080; } in one place and body { background-color: 'white'; } in another), then the rule that comes _last_ will override the others. So if you have any custom CSS that you want to override the default style from the jspsych.css file, then you should make sure that your custom CSS comes _after_ the jspsych.css file.
Hope that helps! Feel free to post some example code if it's still not working.

**Reply by ontinglo** [reply]
thank you so much becky !

**Comment by jwylie21** [comment]
How do you make only a single page a different background color? I need to be able to switch back and forth from white to black backgrounds.

Any ideas would be greatly appreciated!

**Reply by psychbruce** [reply]
There are many ways to switch background color back and forth. (Always remember that it is NOT a good idea to do this in a CSS file, unless you won't change any CSS style during the experiment. As mentioned in my last reply, you should set HTML DOM style by using the "document" object in JavaScript.)

### Method 1: Use the jsPsych plugin "call-function" and add the elements as blocks in your timeline.
```js
var set_html_style_1 = {
    type: 'call-function',
    func: function() {
        document.body.style.backgroundColor = 'white' // background color
        document.body.style.color = 'black' // font color
        document.body.style.fontSize = '20pt'
        document.body.style.fontFamily = 'Arial'
    }
}

var set_html_style_2 = {
    type: 'call-function',
    func: function() {
        document.body.style.backgroundColor = 'black'
        document.body.style.color = 'white'
    }
}
```

### Method 2: Define them in functions and add the functions as parameters in your block.
```js
function set_html_style_1() {
    document.body.style.backgroundColor = 'white' // background color
    document.body.style.color = 'black' // font color
}

function set_html_style_2() {
    document.body.style.backgroundColor = 'black'
    document.body.style.color = 'white'
}

var your_block = {
    type: 'html-keyboard-response',
    on_start: set_html_style_1,
    stimulus: 'Welcome',
    choices: [' '],
    on_finish: set_html_style_2
}
```

### Method 3: Just directly set the "document" object in your block.
```js
var your_block = {
    type: 'html-keyboard-response',
    on_start: function() {
        document.body.style.backgroundColor = 'white'
        document.body.style.color = 'black'
    },
    stimulus: 'Welcome',
    choices: [' '],
    on_finish: function() {
        document.body.style.backgroundColor = 'black'
        document.body.style.color = 'white'
    }
}
```

I suggest using **Method 2**, because once you define the functions, you can **re-use** them anywhere.

**Reply by jwylie21** [reply]
Thank you!!!
